### ISSUE TITLE
Restart doesn't work with `cargo run` that starts a server

### ISSUE THREAD

**Issue by robinst** [mattgreen/watchexec#16]
With a simple Iron server project that listens on a port, the restarting doesn't work (on MacOS):

```
$ watchexec --restart cargo run
// made changes here
   Compiling iron-example v0.1.0 (file:///Users/robinst/Projects/iron-example)
    Finished debug [unoptimized + debuginfo] target(s) in 8.49 secs
     Running `target/debug/iron-example`
// made changes here
   Compiling iron-example v0.1.0 (file:///Users/robinst/Projects/iron-example)
    Finished debug [unoptimized + debuginfo] target(s) in 6.46 secs
     Running `target/debug/iron-example`
thread 'main' panicked at 'Unable to start server: Io(Error { repr: Os { code: 48, message: "Address already in use" } })', ../src/libcore/result.rs:788
note: Run with `RUST_BACKTRACE=1` for a backtrace.
error: Process didn't exit successfully: `target/debug/iron-example` (exit code: 101)
^C
[1]    56690 interrupt  watchexec --restart cargo run
```

The reason for this is that the restart didn't successfully kill the subprocess that `cargo run` started, hence "Address already in use".

Note that `cargo watch` has exactly the same problem: https://github.com/passcod/cargo-watch/issues/25

The cargo developers recommend using a process group and then killing the process group to make sure all subprocesses are killed: https://github.com/rust-lang/cargo/pull/2818#issuecomment-250534859

**Comment by mattgreen** [comment]
As of f3c6df8845ed13e231cd9cbd88afdebe5fd97569, we use `setpgid()`, `killpg()`, and `waitpid()` to handle launching the child process in a new process group on POSIX systems.

We still need to handle when `SIGINT` is sent to watchexec by killing the process group.

**Comment by robinst** [comment]
That was quick :). Tried out watchexec 1.1.1, but restarting still runs into the same problem. Was it expected to be fixed with 1.1.1 (not sure from your description)?

**Comment by mattgreen** [comment]
Target 1.2 with this feature. 1.1.1 was a bugfix for 1.1.

Trying to get signal handling in so we can shutdown cleanly and destroy the process group at exit now. Unfortunately, Rust's signal situation is pretty barebones at the moment.

**Comment by mattgreen** [comment]
This landed in 1.2.0. Give it a try!

**Comment by robinst** [comment]
Yes, works nicely now :)! 🎉
